code/modules/vehicles/mecha/mecha_movement.dm

```diff
@@ -4,9 +4,24 @@
 	for(var/mob/living/occupant as anything in occupants)
 		occupant.setDir(newdir)
 
+///Called when the mech moves
+/obj/vehicle/sealed/mecha/proc/on_move()
+	SIGNAL_HANDLER
+
+	collect_ore()
+	play_stepsound()
+
+///Collects ore when we move, if there is an orebox and it is functional
+/obj/vehicle/sealed/mecha/proc/collect_ore()
+	if(!HAS_TRAIT(src, TRAIT_OREBOX_FUNCTIONAL) || !ore_box)
```

Comment:
```suggestion
	if(isnull(ore_box) || !HAS_TRAIT(src, TRAIT_OREBOX_FUNCTIONAL))
```

`isnull()` is more explicit, and shortcircuiting like this is arguably faster than the list lookup we do in `HAS_TRAIT()`, as well as just looking nicer overall.